forc-plugins/forc-client/src/util/pkg.rs

```diff
@@ -4,10 +4,25 @@ use anyhow::Result;
 use forc_pkg::{self as pkg, manifest::ManifestFile, BuildOpts, BuildPlan};
 use pkg::{build_with_options, BuiltPackage, PackageManifestFile};
 
+#[derive(Clone, Debug)]
+pub struct BuiltPackageWithManifest(Arc<BuiltPackage>, PackageManifestFile);
+
+impl BuiltPackageWithManifest {
+    /// Returns an immutable reference into the Arc<BuiltPackage>.
+    pub fn built_package(&self) -> &Arc<BuiltPackage> {
+        &self.0
+    }
+
+    /// Returns an immutable reference into the PackageManifestFile.
+    pub fn package_manifest_file(&self) -> &PackageManifestFile {
+        &self.1
+    }
+}
+
```

Comment:
This is a nice abstractions but it made me think a little and I think returning `Vec` of `(PackageManifestFile, Arc<BuiltPackage>)` might be unnecessary at the first place from `built_pkg_with_manifest` function. Every built package contains a package descriptor which contains the related package manifest so returning a tuple is kind of a data duplication. 

We can also get this in and think about it later, maybe refactor it slightly such that we do not need to return duplicated package manifests.

Reply:
Good catch, going to create a new issue to address this